File: trunk/sl-maxi/SilverlightMaxi/SilverlightMaxi/Photo.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using System.IO;
using System.Linq;
using System.Xml;
using System.Xml.Linq;
using System.Collections.Generic;
using System.Collections;

namespace SilverlightMaxi
{

    public class Photo
    {
        private long pid;
        private long time;
        private long taker;
        private String url;
        private IEnumerable<long> taggedList;

        public void setTaggedList(IEnumerable<long> taggedList)
        {
            this.taggedList = taggedList;
        }
        public IEnumerable<long> getTaggedList()
        {
            return taggedList;
        }
        public void setPid(long pid)
        {
            this.pid = pid;
        }
        public long getPid()
        {
            return pid;
        }
        public void setTime(long time)
        {
            this.time = time;
        }
        public long getTime()
        {
            return time;
        }
        public void setTaker(long taker)
        {
            this.taker = taker;
        }
        public long getTaker()
        {
            return taker;
        }
        public void setUrl(String url)
        {
            this.url = url;
        }
        public String getUrl()
        {
            return url;
        }

        static public IOrderedEnumerable<Photo> getPhotoList(String filename, Object o)
        {
            string graphResource = filename;
            //string graphResource = "Graphite.Demo.friends.xml";
            System.IO.Stream s = o.GetType().Assembly.GetManifestResourceStream(graphResource);

            XDocument xml = XDocument.Load(XmlReader.Create(s));
            XElement elementRoot = xml.Root; // Not sure if this is necessary
            var elements = from element in elementRoot.Descendants("photo")
                           select new
                           {
                               pid = (long)element.Element("photoid"),
                               time = (long)element.Element("time"),
                               taker = (long)element.Element("taker"),
                               url = (String)element.Element("url"),
                               taggedlist = from friend in element.Element("tagged").Descendants()
                                            select (long)element.Element("uid"),
                           };

            List<Photo> list = new List<Photo>();
            foreach (var el in elements)
            {
                Photo p = new Photo();
                p.setPid(el.pid);
                p.setTaker(el.taker);
                p.setUrl(el.url);
                p.setTime(el.time);
                p.setTaggedList(el.taggedlist);
                list.Add(p);
            }
            return list.OrderBy(p => p.getTime());
            
//            return list.Values.ToList();// ToList();
        }
    }
}

File: trunk/sl-maxi/Facebook Downloader/Photo.java

import java.util.*;
import net.sf.fb4j.model.PhotoTag;

public class Photo implements java.io.Serializable {

	private long taker;
	private Date time;
	private long pid;
	private HashSet<Long> members;
	private String url;

	private static final long serialVersionUID = -1727565347711332372L;


	public String toXml(){
		String ret="<photo>\r\n";
		ret+="<taker>"+taker+"</taker>\r\n";
		ret+="<time>"+time.getTime()+"</time>\r\n";
		ret+="<photoid>"+pid+"</photoid>\r\n";
		ret+="<url>"+url+"</url>\r\n";

		ret+="<tagged>\r\n";
		for(Long l: members){
			ret+="<uid>"+l+"</uid>\r\n";
		}
		ret+="</tagged>\r\n";

		ret+="</photo>\r\n";
		return ret;
	}

	public Photo(){}

	public void setTaker(long taker) {
		this.taker = taker;
	}

	public void setTime(Date time) {
		this.time = time;
	}

	public void setPid(long pid) {
		this.pid = pid;
	}

	public void setMembers(HashSet<Long> members) {
		this.members = members;
	}

	public Photo(Date time, long taker, long pid){
		this.taker = taker;
		this.time = time;
		this.pid = pid;
		members = new HashSet<Long>();
		members.add(taker);
	}

	public long getTaker(){
		return taker;
	}
	public Date getTime(){
		return time;
	}
	public long getPid(){
		return pid;
	}
	public HashSet<Long> getMembers(){
		return members;
	}

	public void addMember(long m){
		members.add(m);
	}

	public boolean equals(Object o){
		if(o instanceof Photo){
			return ((Photo)o).pid==pid;
		}
		return false;
	}
	public int hashCode(){
		return Long.valueOf(pid).hashCode();
	}
}
